src/affine_dimensions.jl

```diff
@@ -0,0 +1,466 @@
+    @assert isone(ustrip(qout)) "You passed a quantity with a non-unit value to uconvert."
+    dout = dimension(qout)
+    dimension(q) == basedim(dout) || throw(DimensionError(q, qout))
+    vout = (ustrip(q) .- offset(dout))./scale(dout)
+    return QuantityArray(vout, dout, quantity_type(q))
+end
+
+# Conversions for AbstractAffineDimensions |> AbstractSymbolicDimensions =======================================================
+function uconvert(qout::UnionAbstractQuantity{<:Any, <:AbstractSymbolicDimensions}, qin::AbstractQuantityOrArray{<:Any, <:AbstractAffineDimensions})
+    uconvert(qout, siunits(qin))
+end
+
+# Conversions for AbstractSymbolicDimensions |> AbstractAffineDimensions   =======================================================
+function uconvert(qout::UnionAbstractQuantity{<:Any,<:AbstractAffineDimensions}, qin::AbstractQuantityOrArray{<:Any, <:AbstractSymbolicDimensions})
+    uconvert(qout, siunits(qin))
+end
+
+# Conversions for AbstractAffineDimensions |> AbstractAffineDimensions =======================================================
+function uconvert(qout::UnionAbstractQuantity{<:Any, <:AbstractAffineDimensions}, qin::AbstractQuantityOrArray{<:Any, <:AbstractAffineDimensions})
+    uconvert(qout, siunits(qin))
+end
+
+# Multiplication and division of AffineDimensions ===============================================================
+function Base.:*(l::AffineDimensions, r::AffineDimensions) 
+    assert_no_offset(l)
+    assert_no_offset(r)
+    return AffineDimensions(
+        scale  = scale(l)*scale(r),
+        offset = offset(l),
+        basedim = basedim(l)*basedim(r)
+    )
+end
+
+function Base.:/(l::AffineDimensions, r::AffineDimensions) 
+    assert_no_offset(l)
+    assert_no_offset(r)
+    return AffineDimensions(
+        scale  = scale(l)/scale(r),
+        offset = offset(l),
+        basedim = basedim(l)/basedim(r)
+    )
+end
+
+# Exponentiation ===============================================================
+function Base.:^(l::AffineDimensions{R}, r::Number) where {R}
+    assert_no_offset(l)
+    return AffineDimensions(
+        scale   = scale(l)^r,
+        offset  = offset(l),
+        basedim = basedim(l)^tryrationalize(R, r)
+    )
+end
+
+function Base.:^(l::AffineDimensions{R}, r::Integer) where {R}
+    assert_no_offset(l)
+    return AffineDimensions(
+        scale   = scale(l)^r,
+        offset  = offset(l),
+        basedim = basedim(l)^tryrationalize(R, r)
+    )
+end
+
+function Base.:inv(l::AffineDimensions{R}) where {R}
+    assert_no_offset(l)
+    return AffineDimensions(
+        scale   = inv(scale(l)),
+        offset  = offset(l),
+        basedim = inv(basedim(l))
+    )
+end
+
+# Operations on self-values ======================================================================================
+function _no_offset_expand(q::Q) where {T, R, D<:AbstractAffineDimensions{R}, Q<:UnionAbstractQuantity{T,D}}
```

Comment:
Current style is no spaces in type expression, so this should be
```suggestion
function _no_offset_expand(q::Q) where {T,R,D<:AbstractAffineDimensions{R},Q<:UnionAbstractQuantity{T,D}}
```